ds0_updater.py

```diff
@@ -5,13 +5,36 @@
 import logger
 import conf
 
+# keep it super simple for now because we don't really implement --allowed-keys yet.
+def verify_latest_commit():
+    allowed_keys = ds0_utils.allowed_keys()
+
+    # ensure the latest commit is signed and verified
+    while True:
+        for key in allowed_keys:
+            logger.exe('gpg --import ' + key['file'], expectError=True)
+        output = logger.exe('git log --pretty="format:%G?" --show-signature HEAD^..HEAD')
+
+        if "Can't check signature" in output[0]:
+            logger.info('gpg keys cleared on startup. Trying again...')
+            continue
+
+        verified = False
+        for key in allowed_keys:
+            if key['rsa_check'] in output[0] and key['signature_check'] in output[0]:
+                verified = True
+        if not verified:
+            logger.error('Scripts using a non-signed commit. Please ensure commit is valid.')
+            logger.error('    If it was a missed signature, feel free to open a ticket at https://github.com/riptano/ComboAMI.')
+        break
+
 # Update the AMI codebase if it's its first boot
 if not conf.get_config("AMI", "CompletedFirstBoot"):
     # check if a specific commit was requested
     force_commit = ds0_utils.required_commit()
 
     # update the repo
-    logger.exe('git pull')
+    logger.exe('git pull ' + ds0_utils.repository())
```

Comment:
My thinking was that it makes sense to stick to elements that users of git (and I assume most version control systems) are familiar with: repository, branch, tag, and commit id. So the interface (options passed in user data) should take one or more of them.

If we want to stick to one option, maybe another direction to consider is [what `npm` uses](https://docs.npmjs.com/files/package.json#git-urls-as-dependencies) for dependencies. It could have a default like:
```
--repository https://github.com/riptano/ComboAMI#2.5
```
When used to replace `--forcecommit`:
```
--repository https://github.com/riptano/ComboAMI#33f501117ba3c4238c28606463315063b5afb68d
```
When used to point to a custom repository:
```
--repository https://github.com/elad/ComboAMI#my_branch_or_tag
```

I would actually prefer it, especially given there's prior art and logic we can borrow for the code. What do you think?